### PR TITLE
Reactor Integration

### PR DESCRIPTION
**Pull request by Cotel**
Integration with http://projectreactor.io/

Still WIP

### CODE REVIEW

**Comment by raulraja** [comment]
@Cotel Looking good so far!, let us know when this is ready for final review.

**Comment by sbuettner** [comment]
Will there also be support for `Mono`?

**Comment by Cotel** [comment]
@sbuettner I'm working on `Mono` instances too 😃

**Comment by sbuettner** [comment]
@Cotel Thats cool, thanks for response.

**Comment by raulraja** [comment]
Ping someone when this is ready for review, you may have to change stuff around once you sync with master

**Comment by Cotel** [comment]
I think this is ready now @pakoito @raulraja

File: modules/effects/arrow-effects-reactor/src/test/kotlin/arrow/effects/MonoKTests.kt [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+  init {
+    testLaws(
+        FunctorLaws.laws(MonoK.functor(), { MonoK.just(it) }, EQ()),
+        ApplicativeLaws.laws(MonoK.applicative(), EQ()),
+        MonadLaws.laws(MonoK.monad(), EQ()),
+        MonadErrorLaws.laws(MonoK.monadError(), EQ(), EQ(), EQ()),
+        ApplicativeErrorLaws.laws(MonoK.applicativeError(), EQ(), EQ(), EQ()),
+        MonadSuspendLaws.laws(MonoK.monadDefer(), EQ(), EQ(), EQ()),
+        AsyncLaws.laws(MonoK.async(), EQ(), EQ(), EQ()),
+        AsyncLaws.laws(MonoK.effect(), EQ(), EQ())
```
Comment (pakoito) [inline modules/effects/arrow-effects-reactor/src/test/kotlin/arrow/effects/MonoKTests.kt]: These laws are duplicated. AsyncLaws implies the other typeclass laws are checked too. And our typeclasses inherit from one another so we're safe. Basically, do the same as FluxK in here :D

File: modules/docs/arrow-docs/docs/docs/README.md [hunk 2]
```diff
@@ -56,6 +56,7 @@ dependencies {
     compile 'io.arrow-kt:arrow-mtl:0.7.2' //optional
     compile 'io.arrow-kt:arrow-effects:0.7.2' //optional
     compile 'io.arrow-kt:arrow-effects-rx2:0.7.2' //optional
+    compile 'io.arrow-kt:arrow-effects-reactor:0.7.2' //optional
```
Comment (pakoito) [inline modules/docs/arrow-docs/docs/docs/README.md]: Make sure to add this line to the README in the root of the repo. It isn't the same as this one, sadly.

File: modules/effects/arrow-effects-reactor/src/test/kotlin/arrow/effects/MonoKTests.kt [hunk 3]
```diff
@@ -64,14 +69,7 @@ class MonoKTest : UnitSpec() {
...
     testLaws(
-        FunctorLaws.laws(MonoK.functor(), { MonoK.just(it) }, EQ()),
-        ApplicativeLaws.laws(MonoK.applicative(), EQ()),
-        MonadLaws.laws(MonoK.monad(), EQ()),
-        MonadErrorLaws.laws(MonoK.monadError(), EQ(), EQ(), EQ()),
-        ApplicativeErrorLaws.laws(MonoK.applicativeError(), EQ(), EQ(), EQ()),
-        MonadSuspendLaws.laws(MonoK.monadDefer(), EQ(), EQ(), EQ()),
-        AsyncLaws.laws(MonoK.async(), EQ(), EQ(), EQ()),
-        AsyncLaws.laws(MonoK.effect(), EQ(), EQ())
+        AsyncLaws.laws(MonoK.async(), EQ(), EQ())
```
[the remark this answers is not in the document]
Comment (Cotel) [reply modules/effects/arrow-effects-reactor/src/test/kotlin/arrow/effects/MonoKTests.kt]: MonoK has no instance for Traverse and SingleK does not have it either 🤔 .

[the remark this answers is not in the document]
Comment (pakoito) [reply modules/effects/arrow-effects-reactor/src/test/kotlin/arrow/effects/MonoKTests.kt]: Okey dokey then! :D

**Comment by pakoito** [comment]
Nnnnnnice, good work :D

**Pull request merged**